### DISCUSSION TITLE
How To: Store User's Data

### DISCUSSION THREAD

**Discussion by cdt-eth** [supabase/supabase#2335] [Questions] [answered]
I have Auth set up on my personal project, a Disney Plus Clone ([GitHub](https://github.com/christiandavidturner/Disney-Plus-Plus) | [Deployed Site](https://disnyplus.plus/)), and now I'm trying to "add movies" to a user's watchlist.

This is my first time working with a Database & Users on a project so I'm not 100% sure how to proceed. But I'm have a Result Page for each movie/show. There's a `+` button with a `handleClick` function that currently just prints out said movie/show's ID. 

My guess is what I _should_ do is:
1) once logged in, push said movie/show `id` to Supabase DB, for said specific user.
<img width="722" alt="Screen Shot 2021-07-12 at 9 38 48 AM" src="https://user-images.githubusercontent.com/26611339/125297151-fd082880-e2f4-11eb-81b4-1d78e46cce10.png">

This is the way the production Disney+ Watchlist page looks. 
<img width="1266" alt="Screen Shot 2021-07-12 at 9 43 09 AM" src="https://user-images.githubusercontent.com/26611339/125297809-93d4e500-e2f5-11eb-8ffb-af381be82d51.png">

2) So on this Watchlist page, I'd fetch the `id`'s from said specific user and display it here (I have a `Result` component I'll map the `id`'s from).

Sounds easy enough, not sure the Supabase method of _doing_ this but I'm moreso looking for validation of the thought process since again I'm new to storing user data & database storage.

**Comment by burggraf** [comment, marked as answer]
Your app looks fantastic, Christian!  You're approaching things correctly.  Just get the user's `id`, which is in the `auth.users` table.  You'll probably want a table in the `public` schema to store your user data, just put the user's `id` in each record of that table so it's easy to locate it.  You should find working with the database in Supabase very straighforward.  If you have any questions, check out the documentation here [https://supabase.io/docs/reference/javascript/select](https://supabase.io/docs/reference/javascript/select) or ask questions here on Github.

**Reply by cdt-eth** [reply]
thanks mark! i appreciate it.

i got it working and now can `insert` & `delete` items from a user's DB. now i need stored state, so looks like it's time to implement Redux for the first time haha

**Reply by burggraf** [reply]
Great!  Hope you find working with Supabase as fun as I have.

**Reply by borngraced** [reply]
> thanks mark! i appreciate it.
> 
> i got it working and now can `insert` & `delete` items from a user's DB. now i need stored state, so looks like it's time to implement Redux for the first time haha

Don't forget to checkout Context API and useReducer buddy!

**Reply by cdt-eth** [reply]
thanks @burggraf & @Samuelonoja 

I'm familiar with Context, but not useReducer, heard of it though. So say I have a signed in user and they add a movie to their watchlist. It's stored in the database but I want some UI/state to update and persist to tell the user that said movie is in the database.

Likewise, they should be able to click the button to delete from the database (which i've set up with Supabase), but I then need the UI/state to update/revert to another place. Again, it needs to persists if they log out and log back in or navigate to another page and then come back etc.

That's why I assumed I'd need Redux.
